lib/IDE/CodeCompletionResult.cpp

```diff
@@ -71,18 +75,31 @@ ContextFreeCodeCompletionResult::createLiteralResult(
   return new (Sink.getAllocator()) ContextFreeCodeCompletionResult(
       CodeCompletionResultKind::Literal, static_cast<uint8_t>(LiteralKind),
       CodeCompletionOperatorKind::None,
-      /*IsSystem=*/false, CompletionString, /*ModuleName=*/"",
+      /*IsSystem=*/false, /*IsAsync=*/false, CompletionString,
+      /*ModuleName=*/"",
       /*BriefDocComment=*/"",
       /*AssociatedUSRs=*/{}, ResultType, ContextFreeNotRecommendedReason::None,
       CodeCompletionDiagnosticSeverity::None, /*DiagnosticMessage=*/"",
-      getCodeCompletionResultFilterName(CompletionString, Sink.getAllocator()));
+      getCodeCompletionResultFilterName(CompletionString, Sink.getAllocator()),
+      /*NameForDiagnostics=*/"");
+}
+
+static NullTerminatedStringRef
+getDeclNameForDiagnostics(const Decl *D, CodeCompletionResultSink &Sink) {
+  if (auto VD = dyn_cast<ValueDecl>(D)) {
+    llvm::SmallString<64> Scratch;
+    return NullTerminatedStringRef(VD->getName().getString(Scratch),
+                                   Sink.getAllocator());
```

Comment:
An implementation equivalent to the current implementation would be:
```suggestion
    llvm::SmallString<64> Scratch;
    llvm::raw_svector_ostream OS(Scratch);
    VD->getName().printPretty(OS);
    return NullTerminatedStringRef(Scratch, Sink.getAllocator());
```